src/main/java/com/yelp/nrtsearch/server/luceneserver/RegisterFieldsHandler.java

```diff
@@ -127,6 +132,18 @@ public FieldDefResponse handle(final IndexState indexState, FieldDefRequest fiel
     return reply;
   }
 
+  private void verifyOnlyOneIdFieldExists(Map<String, FieldDef> pendingFieldDefs, FieldDef fieldDef)
```

Comment:
This will also need to make sure there is not an _id field already registered in the index state.